# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/HyperTextAccessibleBase.h
# Commit: 5e92b9c7d479
# Full Hash: 5e92b9c7d4794af1d7ab0fb5c278c2e681a18fc6
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1811092
# File Overlap Count: 1
# Description:
#   Bug 1811092 - P1: Add new async add/set text selection message. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167201
# ==============================================================================

diff -r 7cebaa85f4ff -r 5e92b9c7d479 accessible/basetypes/HyperTextAccessibleBase.h
--- a/accessible/basetypes/HyperTextAccessibleBase.h	Fri Jan 27 20:03:06 2023 +0000
+++ b/accessible/basetypes/HyperTextAccessibleBase.h	Fri Jan 27 20:15:11 2023 +0000
@@ -203,6 +203,22 @@
   virtual bool SelectionBoundsAt(int32_t aSelectionNum, int32_t* aStartOffset,
                                  int32_t* aEndOffset);
 
+  /**
+   * Changes the start and end offset of the specified selection.
+   * @return true if succeeded
+   */
+  // TODO: annotate this with `MOZ_CAN_RUN_SCRIPT` instead.
+  MOZ_CAN_RUN_SCRIPT_BOUNDARY virtual bool SetSelectionBoundsAt(
+      int32_t aSelectionNum, int32_t aStartOffset, int32_t aEndOffset);
+
+  /**
+   * Adds a selection bounded by the specified offsets.
+   * @return true if succeeded
+   */
+  bool AddToSelection(int32_t aStartOffset, int32_t aEndOffset) {
+    return SetSelectionBoundsAt(-1, aStartOffset, aEndOffset);
+  }
+
  protected:
   virtual const Accessible* Acc() const = 0;
   Accessible* Acc() {